### PR TITLE
docs: Add a troubleshoot section in hacking

### PR DESCRIPTION
**Pull request by pawamoy**
Following the issues and solutions I went through while setting up a development/testing environment (see #1425)

### CODE REVIEW

File: docs/manual/hacking.rst [hunk 1]
```diff
@@ -18,6 +18,13 @@ On Ubuntu, if testing on Python 3, this can be done with:
 
     sudo apt-get install python3-pytest xserver-xephyr x11-apps
 
+On ArchLinux, the X11 requirements are installed with:
+
+.. code-block:: bash
+
+    sudo pacman -S xorg-xrandr xorg-xcalc xorg-xeyes xorg-xclock
+
+
```
Comment (pawamoy) [inline docs/manual/hacking.rst]: Not sure if you're willing to maintain these distro-dependent installation methods. There already was Ubuntu though, so I thought why not adding Arch.

File: docs/manual/hacking.rst [hunk 2]
```diff
@@ -57,6 +65,8 @@ In practice, the development cycle looks something like this:
 #. run entire test suite: ``pytest``
 #. commit
 
+You don't need to run the xephyr script in order to run the tests.
```
Comment (pawamoy) [inline docs/manual/hacking.rst]: This is what I initially thought, so it might be worth to note.

**Comment by pawamoy** [comment]
Build timeout again :confused:

File: docs/manual/hacking.rst (hunk 2 above)
Comment (flacjacket) [reply docs/manual/hacking.rst]: You could note that the test runner will launch its own x instances.

File: docs/manual/hacking.rst [hunk 3]
```diff
@@ -153,3 +162,89 @@ Here are a number of resources that may come in handy:
...
+
+If it happens, it might be because the `cairocffi` and `xcffib` dependencies
+were installed in the wrong order.
+
+To fix this:
+
+1. uninstall them from your environment: with ``pip uninstall cairocffi xcffib``
+   if using a virtualenv, or with your system package-manager if you installed
+   the development version of Qtile system-wide.
+#. re-install them sequentially (again, with pip or with your package-manager)::
```
Comment (flacjacket) [inline docs/manual/hacking.rst]: If it is a package manager, it may be up to fixing the packaging of cairocffi in the relevant distro.

Comment (pawamoy) [reply docs/manual/hacking.rst]: I added this as a note just below.

**Comment by pawamoy** [comment]
GitHub says this branch has no conflicts with the base branch :confused:. Should I still checkout this local branch and rebase it on master?

**Comment by pawamoy** [comment]
Hmm I think it's good now, I rebased anyway.

**Comment by pawamoy** [comment]
Once again, the CI failed because of a timeout on the pypy3 job, not sure why :slightly_smiling_face:

**Comment by ramnes** [comment]
I've just allowed failures for the PyPy3 build, simply rebase and this won't be a problem anymore.

**Comment by pawamoy** [comment]
It's rebased :+1:

**Comment by flacjacket** [comment]
Thanks!

**Pull request merged**